src/rng.rs

```diff
@@ -393,14 +393,20 @@ impl Fill for [u8] {
     }
 }
 
-macro_rules! impl_fill {
+// This macro is unsafe to call: target types must support transmute from
+// random bits (i.e. all bit representations are valid).
```

Comment:
To be precise: In order to call this for a type, `T`, it must be the case that `mem::transmute::<[u8; size_of::<T>()], T>(...)` is sound for all values. Note that this precludes uninitialized bytes.

Also, to be pedantic, this should be a doc comment with a `# Safety` section.

Reply:
I added a doc comment with a safety section, using the wording representation wording @RalfJung suggested above.